compiler/noirc_evaluator/src/ssa/interpreter/mod.rs

```diff
@@ -875,15 +875,33 @@ impl<'ssa, W: Write> Interpreter<'ssa, W> {
         result: ValueId,
         side_effects_enabled: bool,
     ) -> IResult<()> {
-        let element = if side_effects_enabled {
-            let array = self.lookup_array_or_slice(array, "array get")?;
-            let index = self.lookup_u32(index, "array get index")?;
-            let index = index - offset.to_u32();
-            array.elements.borrow()[index as usize].clone()
-        } else {
+        let array = self.lookup_array_or_slice(array, "array get")?;
+        let index = self.lookup_u32(index, "array get index")?;
+        let mut index = index - offset.to_u32();
+        // An array_get with false side_effects_enabled is replaced
+        // by a load at a valid index during acir-gen.
+        if !side_effects_enabled && array.elements.borrow().len() <= index as usize {
```

Comment:
It looks like this still doesn't quite match acir. Acir does:
```rust
if array length is 0 {
    assert(!predicate)
} else {
    array[i * predicate]
}
```
Where this does:
```rust
if !predicate && array len <= index {
    if no array elements {
        define(result, 0);
        return;
    }
    index = find_index_with_matching_type(array, result_type);
}
array[index]
```

Reply:
Indeed, in a previous comment I said:
> I think it is not worth to try to get the exact same behaviour

The behaviour in acir-gen is quite complex, but in the case of an empty array it will be similar: use 0 value is predicate is false, and fail if the predicate is true.
Should I replicate the full behaviour?

Reply:
The problem is that acir-gen behaviour depends on constant vs non-constant values, while the ssa interpreter depends on ssa values, which can become constant on a later stage (e.g after optimisations), so the behaviour would not be identical depending on which ssa representation is used.